### PR TITLE
Libtomcrypt: use buf_compare_ct instead of mem_neq

### PR DESCRIPTION
**Pull request by pascal-brand-st-dev**
Signed-off-by: Pascal Brand <pascal.brand@st.com>

### CODE REVIEW

File: core/lib/libtomcrypt/src/misc/sub.mk [hunk 1]
```diff
@@ -1,6 +1,6 @@
 srcs-y += burn_stack.c
 srcs-y += error_to_string.c
-srcs-y += mem_neq.c
+# srcs-y += mem_neq.c
```
Comment (jforissier) [inline core/lib/libtomcrypt/src/misc/sub.mk]: Just remove the line

File: core/lib/libtomcrypt/include/tomcrypt_custom.h [hunk 2]
```diff
@@ -80,6 +80,10 @@
    #endif
 #define XMEMCMP  memcmp
 #endif
+#ifndef XMEM_NEQ
+#include <string_ext.h>
+#define XMEM_NEQ  buf_compare_ct
+#endif
```
Comment (jforissier) [inline core/lib/libtomcrypt/include/tomcrypt_custom.h]: How is XMEM_NEQ useful here? You could just `#define mem_neq buf_compare_ct` here. Advantage: no change to the .c files.

Comment (pascal-brand-st-dev) [inline core/lib/libtomcrypt/include/tomcrypt_custom.h]: .c file has been modified this way in libtomcrypt

File: core/lib/libtomcrypt/src/misc/sub.mk (hunk 1 above)
Comment (pascal-brand-st-dev) [inline core/lib/libtomcrypt/src/misc/sub.mk]: ok

File: core/lib/libtomcrypt/include/tomcrypt_custom.h (hunk 2 above)
Comment (jforissier) [inline core/lib/libtomcrypt/include/tomcrypt_custom.h]: OK then.

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
This change adds a non-standard dependency to LTC which will make it hard to upstream. Other than that:
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```

**Pull request merged**